src/frontend/packages/core/src/shared/services/cloud-foundry-user-provided-services.service.ts

```diff
@@ -113,7 +123,15 @@ export class CloudFoundryUserProvidedServicesService {
     guid: string,
     data: Partial<IUserProvidedServiceInstanceData>,
   ): Observable<RequestInfoState> {
-    const updateAction = new UpdateUserProvidedServiceInstance(cfGuid, guid, data, this.serviceInstancesEntityConfig);
+    const userProvidedServiceEntity = entityCatalogue.getEntity(CF_ENDPOINT_TYPE, userProvidedServiceInstanceEntityType);
+    const actionBuilder = userProvidedServiceEntity.actionOrchestrator.getActionBuilder('update');
+    // TODO kate verify OK
```

Comment:
TODO

Reply:
I totally agree with some kind of helper but, for the case where we only use the catalogue entity to get the action, we could chain all of the functions together. That would get you to at least something like: 

```
const action = entityCatalogue
      .getEntity(CF_ENDPOINT_TYPE, serviceBindingEntityType)
      .actionOrchestrator
      .getActionBuilder('getMultiple')(
        cfGuid,
        createEntityRelationPaginationKey(serviceEntityType, serviceInstanceGuid)
      );
```